data-collector/internal/client/client.go

```diff
@@ -0,0 +1,58 @@
+package client
+
+import (
+	"io"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+type Client interface {
+	Get(string) ([]byte, int, error)
+	Post(string, io.Reader) ([]byte, int, error)
+}
+
+type StandardClient struct {
+	baseURL string
+
+	httpClient *http.Client
+}
+
+var (
+	_ Client = (*StandardClient)(nil)
+)
+
+func NewClient() *StandardClient {
+	return &StandardClient{
+		baseURL: "https://localhost:3000/api/v1/",
```

Comment:
This API doesn't actually exist yet. We are sending all required information so this portion is done.